bootstrap.py2

```diff
@@ -0,0 +1,136 @@
+#!/opt/pypy/bin/pypy
+
+import decimal
+import json
+import os
+import site
+import sys
+import urllib2 as request
+import time
+
+HANDLER = os.getenv("_HANDLER")
+RUNTIME_API = os.getenv("AWS_LAMBDA_RUNTIME_API")
+
+for path in ["/opt/pypy", "/opt/pypy/site-packages"]:
+    sys.path.insert(0, path)
+    site.addsitedir(path)
+
+if "LAMBDA_TASK_ROOT" in os.environ:
+    sys.path.insert(0, os.environ["LAMBDA_TASK_ROOT"])
+    site.addsitedir(os.environ["LAMBDA_TASK_ROOT"])
+
+
+class LambdaContext(object):
+    def __init__(self, request_id, invoked_function_arn, deadline_ms, trace_id):
+        self.aws_request_id = request_id
+        self.deadline_ms = deadline_ms
+        self.function_name = os.getenv("AWS_LAMBDA_FUNCTION_NAME")
+        self.function_version = os.getenv("AWS_LAMBDA_FUNCTION_VERSION")
+        self.invoked_function_arn = invoked_function_arn
+        self.log_group_name = os.getenv("AWS_LAMBDA_LOG_GROUP_NAME")
+        self.log_stream_name = os.getenv("AWS_LAMBDA_LOG_STREAM_NAME")
+        self.memory_limit_in_mb = os.getenv("AWS_LAMBDA_FUNCTION_MEMORY_SIZE")
+        self.trace_id = trace_id
+        if self.trace_id is not None:
+            os.environ["_X_AMZN_TRACE_ID"] = self.trace_id
+
+    def get_remaining_time_in_millis(self):
+        if self.deadline_ms is not None:
+            return time.time() * 1000 - int(self.deadline_ms)
+
+
+def decimal_serializer(obj):
+    if isinstance(obj, decimal.Decimal):
+        return float(obj)
+    raise TypeError(repr(obj) + " is not JSON serializable")
+
+
+def init_error(message, type):
+    details = {"errorMessage": message, "errorType": type}
+    details = json.dumps(details).encode("utf-8")
+    req = request.Request(
+        "http://{}/2018-06-01/runtime/init/error".format(RUNTIME_API),
+        details,
+        {"Content-Type": "application/json"},
+    )
+    res = request.urlopen(req)
```

Comment:
In `diff bootstrap.py3 bootstrap.py2`

```
-    with request.urlopen(req) as res:
-        res.read()
+    res = request.urlopen(req)
+    res.read()
```

Looks like this change doesn't close the response, and the recommended way seems to be `contextlib.closing` https://docs.python.org/2/library/contextlib.html.

Reply:
Thanks. Fixed. PTAL